src/test/java/io/stargate/sgv2/jsonapi/api/v1/InsertIntegrationTest.java

```diff
@@ -148,6 +148,63 @@ public void insertDocumentWithDate() {
           .body("errors", is(nullValue()));
     }
 
+    @Test
+    public void insertDocumentWithDateDocId() {
+      String json =
+          """
+            {
+              "insertOne": {
+                "document": {
+                  "_id": {"$date": 1672539900000},
+                  "username": "doc_date_user4",
+                  "status": false
+                }
+              }
+            }
+            """;
+
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("status.insertedIds[0]", jsonEquals("{\"$date\":1672539900000}"))
+          .body("data", is(nullValue()))
+          .body("errors", is(nullValue()));
+
+      String expected =
+          """
+            {
+              "_id": {"$date": 1672539900000},
+              "username": "doc_date_user4",
+              "status": false
+            }
+            """;
+
+      String query_json =
```

Comment:
@maheshrajamani This fails, not finding the document. I haven't been able to figure out why (DBFilterBase handling seems fine wrt DocumentIds) so may need help here.

Reply:
@tatu-at-datastax Had to fix in FilterClauseDeserializer to resolve "_id" with date as DocumentId. It was resolving to Date. I have fixed it and checked in to this branch. Let me know if it is fine.